packages/react-compose/src/createClassResolver.ts

```diff
@@ -14,43 +23,74 @@ import { GenericDictionary, ClassDictionary } from './types';
  * Remaining class names would be interpretted as modifiers, applied to the `root` slot when
  * the component `state` contains a truthy matching prop name.
  */
-export const createClassResolver = (classes: ClassDictionary) => (
-  state: GenericDictionary,
-  slots: GenericDictionary,
-): ClassDictionary => {
-  const classMap: GenericDictionary = {};
-  const modifiers: string[] = [];
-
-  // Add the default className to root
-  addClassTo(classMap, 'root', state.className);
-
-  if (classes) {
-    // Iterate through classes
-    Object.keys(classes).forEach((key: string) => {
-      const classValue = classes[key];
-
-      if (classValue) {
-        // If the class is named the same as a slot, add it to the slot.
-        if (slots.hasOwnProperty(key)) {
-          addClassTo(classMap, key, classValue);
-        } else if (key.indexOf('_') >= 0) {
-          // The class is an enum value. Add if the prop exists and matches.
-          const parts = key.split('_');
-          const enumName = parts[0];
-          const enumValue = parts[1];
-
-          state[enumName] === enumValue && modifiers.push(classValue);
-        } else {
-          state[key] && modifiers.push(classValue);
-        }
+export const createClassResolver = (staticClasses: ClassDictionary) => {
+  const cacheMap: RecursiveMemoNode = new Map();
+  let cacheMissCount = 0;
+  let disableCaching = false;
+
+  return (
```

Comment:
So, with static classes, if we know ahead of time the slots, we can actually pull out the deps automatically without them being provided. Basically every key in the class map that isn't a slot would be the key to depend on in `state`

Reply:
we'd just need to know slots at creation time.

Reply:
@dzearing I'm guessing this is what you're doing in #13448.

Reply:
I removed the memo changes from this PR for now